### PR TITLE
[feat] Add ability to pin metrics from UI

### PR DESCRIPTION
**Pull request by roubkar**
Implemented metrics pinning functionality with "optimistic updates".
Pinned metrics are sorted based on their pinning order.

### CODE REVIEW

**Comment by roubkar** [comment]
Resolves https://github.com/aimhubio/aim/issues/1990

**Review by arsengit — APPROVED** [review]

File: aim/web/ui/src/pages/RunDetail/RunDetail.scss [hunk 1]
```diff
@@ -227,7 +227,12 @@
 }
 
 .RunDetailMetricsTab {
-  padding: 1.5rem 0;
+  padding: 0 0 1.5rem;
```
Comment (VkoHov) [inline aim/web/ui/src/pages/RunDetail/RunDetail.scss:230]: Let's change the 1.5rem to $space-lg

File: aim/web/ui/src/pages/RunDetail/types.d.ts [hunk 2]
```diff
@@ -1,4 +1,5 @@
 import { TraceRawDataItem, TraceType } from 'services/models/runs/types';
+import { IPinnedSequence } from 'types/services/models/projects/projectsModel';
```
Comment (VkoHov) [inline aim/web/ui/src/pages/RunDetail/types.d.ts:2]: the linter suggests adding a space between these two.

**Review by VkoHov — APPROVED** [review]
lgtm!

**Pull request merged**